### PR TITLE
Unit and Integration tests

### LINKED ISSUE #7: get a test harness in place for slack bot
need to test this code - what we have so far is a spike

### PR DESCRIPTION
**Pull request by tansaku**
fixes #7

### CODE REVIEW

File: test/controller.test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    var self = this;
+    self.slackId = 'test'
+    self.userName = 'test'
+    self.controller = new botMock.controller(self.slackId, self.userName)
+    self.input = (text)=>{ return [{
+                first: true,
+                user: self.slackId,
+                messages:[{text: text, isAssertion:true }]
+            }] }
+    var logger = console.log
```
Comment (mtc2013) [inline test/controller.test.js]: @tansaku why have this variable?

Comment (tansaku) [reply test/controller.test.js]: not needed - I was trying to stub the console log, but I ended up using winston

**Comment by tansaku** [comment]
@arreche - thanks for the updates - but I wanted the logging in there - you seem to have removed the winston logging without replacing it with anything ...

**Comment by arreche** [comment]
@tansaku Oh! I thought that botkit's logging system was enough.

```javascript
var controller = Botkit.slackbot({
  debug: false
  //include "log: false" to disable logging
  //or a "logLevel" integer from 0 to 7 to adjust logging verbosity
});
```

**Comment by arreche** [comment]
@tansaku Please. Enable TravisCI integration https://travis-ci.org/AgileVentures/async_slack_bot

**Comment by tansaku** [comment]
@arreche TravisCI activated.

and maybe that logging is enough, hmm - I should test ...

**Pull request merged**